llama_index/vector_stores/simple.py

```diff
@@ -103,6 +126,9 @@ def add(
         """Add embedding_results to index."""
         for result in embedding_results:
             self._data.embedding_dict[result.id] = result.embedding
+            self._data.metadata_dict[result.id] = node_to_metadata_dict(
+                result.node, remove_text=True, flat_metadata=True
```

Comment:
note -- setting `flat_metadata=True` will cause an error to be raised if a user stores anything other than a string, int, or float in the metadata (i.e. no lists, objects, etc.)

I think this is a feature that is heavily (ab)used. Anyone we can support all types of metadata? Special handling for iterables, skipping metadata if we fail to filter on it?

Reply:
Fixed this by writing a filter function that only works for list, str, int, float, and bool

Other values are skipped